core/views.py

```diff
@@ -104,7 +104,25 @@ class ReportsView(LoginRequiredMixin, TemplateView):
 
 @login_required
 def entries_csv_export(request):
-    dataset = EntryResource().export()
-    response = HttpResponse(dataset.csv, content_type='text/csv')
-    response['Content-Disposition'] = 'attachment; filename="entries.csv"'
+    QueryDict = request.GET.copy()
+    kwargs = {}
+    for key in ('project', 'project__client', 'user'):
+        value = request.GET.get(key)
+        if value:
+            kwargs[key] = value
+    if QueryDict['export_format'] is not None:
+        formattype = QueryDict['export_format']
+    if kwargs:
+        queryset = Entry.objects.filter(**kwargs)
+        dataset = EntryResource().export(queryset)
+    if formattype == 'csv':
+        response = HttpResponse(dataset.csv, content_type='text/csv')
+        response['Content-Disposition'] = 'attachment; filename="entries.csv"'
+    elif formattype == 'xls':
+        response = HttpResponse(dataset.xls, content_type='application/vnd.ms-excel')
+        response['Content-Disposition'] = 'attachment; filename="entries.xls"'
+    elif formattype == 'json':
+        response = HttpResponse(dataset.json, content_type='application/json')
+        response['Content-Disposition'] = 'attachment; filename="entries.json"'
+    
```

Comment:
The entire if chain can be turned into a really condensed line after you validate the formattype

    response = HttpResponse(getattr(dataset, formattype), type_content='text/'+ formattype)
    response['Content-Disposition'] = 'attachment; filename="entries.' + formattype + '"'